server/api/store_test.go

```diff
@@ -14,54 +14,164 @@
 package api
 
 import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
 	"net/url"
 
 	. "github.com/pingcap/check"
 	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/pd/server"
 )
 
-type testStoreSuite struct{}
-
 var _ = Suite(&testStoreSuite{})
 
-func (s *testStoreSuite) TestUrlStoreFilter(c *C) {
-	stores := []*metapb.Store{
+type testStoreSuite struct {
+	svr       *server.Server
+	cleanup   cleanUpFunc
+	urlPrefix string
+	client    *http.Client
+	stores    []*metapb.Store
+}
+
+func (s *testStoreSuite) SetUpSuite(c *C) {
+	s.stores = []*metapb.Store{
 		{
 			// metapb.StoreState_Up == 0
-			State: metapb.StoreState_Up,
+			Id:      1,
+			Address: "localhost:1",
+			State:   metapb.StoreState_Up,
 		},
 		{
-			State: metapb.StoreState_Up,
+			Id:      4,
+			Address: "localhost:4",
+			State:   metapb.StoreState_Up,
 		},
 		{
-			// metapb.StoreState_Up == 1
-			State: metapb.StoreState_Offline,
+			// metapb.StoreState_Offline == 1
+			Id:      6,
+			Address: "localhost:6",
+			State:   metapb.StoreState_Offline,
 		},
 		{
 			// metapb.StoreState_Tombstone == 2
-			State: metapb.StoreState_Tombstone,
+			Id:      7,
+			Address: "localhost:7",
+			State:   metapb.StoreState_Tombstone,
 		},
 	}
 
-	var table = []struct {
+	s.svr, s.cleanup = mustNewServer(c)
+	mustWaitLeader(c, []*server.Server{s.svr})
+
+	addr := s.svr.GetAddr()
+	httpAddr := mustUnixAddrToHTTPAddr(c, addr)
+	s.urlPrefix = fmt.Sprintf("%s%s/api/v1", httpAddr, apiPrefix)
+
+	mustBootstrapCluster(c, s.svr)
+	for _, store := range s.stores {
+		mustPutStore(c, s.svr, store)
+	}
+	s.client = newUnixSocketClient()
+}
+
+func (s *testStoreSuite) TearDownSuite(c *C) {
+	s.cleanup()
+}
+
+func checkStoresInfo(c *C, ss []*storeInfo, want []*metapb.Store) {
+	c.Assert(len(ss), Equals, len(want))
+	mapWant := make(map[uint64]*metapb.Store)
+	for _, s := range want {
+		if _, ok := mapWant[s.Id]; !ok {
+			mapWant[s.Id] = s
+		}
+	}
+	for _, s := range ss {
+		c.Assert(s.Store.Store, DeepEquals, mapWant[s.Store.Store.Id])
+	}
+}
+
+func (s *testStoreSuite) TestStoresList(c *C) {
+	resp, err := s.client.Get(fmt.Sprintf("%s/stores", s.urlPrefix))
```

Comment:
can you extract a function to get the json body and decode to an interface, so many test cases can use it.